internal/command/jsonstate/state.go

```diff
@@ -518,16 +538,19 @@ func SensitiveAsBool(val cty.Value) cty.Value {
 		vals := make([]cty.Value, 0, length)
 		it := val.ElementIterator()
 		for it.Next() {
-			_, v := it.Element()
-			vals = append(vals, SensitiveAsBool(v))
+			kv, ev := it.Element()
+			path := append(path, cty.IndexStep{
+				Key: kv,
+			})
+			vals = append(vals, sensitiveAsBoolWithPathValueMarks(ev, path, pvms))
 		}
 		// The above transform may have changed the types of some of the
 		// elements, so we'll always use a tuple here in case we've now made
 		// different elements have different types. Our ultimate goal is to
 		// marshal to JSON anyway, and all of these sequence types are
 		// indistinguishable in JSON.
 		return cty.TupleVal(vals)
-	case ty.IsMapType() || ty.IsObjectType():
+	case ty.IsMapType():
```

Comment:
Since it is certainly a map, we can simplify the length check starting on line 559. However, I'm curious if that's necessary to separate maps and objects processing here (see comment below)

Reply:
two difference between obj and map
1. length or empty check
2. different path step type